### ISSUE TITLE
Build: Update ShadowJar plugin exclude list

### ISSUE THREAD

**Issue by cpeeyush** [eclipse-dataspaceconnector/DataSpaceConnector#1125]
## Description

Current exclude list of ShadowJar plugin is `exclude("**/pom.properties", "**/pom.xm")` which has a typo in `xml` and it should be instead `exclude("**/pom.properties", "**/pom.xml")`.

It has many occurrences across codebase. One sample is [here in launcher](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/98038f7e6bb101d81f96096a58d5a402f3494cc7/launchers/basic/build.gradle.kts#L45).

## Acceptance Criteria

- [ ] Fixed all occurrences of above mentioned typo in this project.

**Comment by ndr-brt** [comment]
do we need this exclusion at all? I would remove it everywhere

**Comment by cpeeyush** [comment]
Some has opinion to have pom.xml in generated JAR and some has opposite opinion. The argument to not have it in generated JAR says that pom.xml may have some information which you only want to utilize during the build and do not want to revel after that in a generated artifact.

In EDC scenario I tried to extract one pf the generated shadow JAR and due to this typo it has few pom.xml in it.

(As always, I found this discussion+comments on SO https://stackoverflow.com/a/1477687/358193 on the topic.)

I will leave you and other EDC devs to decide what is a better solution here, issue was to just highlight the typo. Thanks
